modules/dnn/src/layers/convolution_layer.cpp

```diff
@@ -682,18 +731,23 @@ class ConvolutionLayerImpl CV_FINAL : public BaseConvolutionLayerImpl
                     for( int ofs0 = stripeStart; ofs0 < stripeEnd; ofs0 += BLK_SIZE )
                     {
                         int ofs, ofs1 = std::min(ofs0 + BLK_SIZE, stripeEnd);
-                        int out_i = ofs0 / outW;
-                        int out_j = ofs0 - out_i * outW;
+
+                        int out_d = ofs0 / (outH * outW);
+                        int out_i = (ofs0 - out_d * outH * outW) / outW;
+                        int out_j = ofs0 % outW;
 
                         // do im2row for a part of input tensor
                         float* rowbuf = rowbuf0;
-                        for( ofs = ofs0; ofs < ofs1; out_j = 0, ++out_i )
+
+                        for( ofs = ofs0; ofs < ofs1; out_d += (out_i + 1) / outH, out_i = (out_i + 1) % outH, out_j = 0 )
```

Comment:
Can you please explain this changes?

I mean I just wanted just to understand them. Previously we had at least one dimension which were incremented by 1 every iteration but now all the dimensions are computed.

Reply:
```
for( ofs = ofs0; ofs < ofs1; out_j = 0, ++out_i ) {
     if (out_i == outH) {
        out_i = 0;
        ++out_d;
    }
    // do something
}

```